focus-android/app/src/main/java/org/mozilla/focus/open/AppAdapter.kt

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.focus.open
+
+import android.content.Context
+import android.content.pm.ActivityInfo
+import android.graphics.drawable.Drawable
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+
+/**
+ * An adapter for displaying app items in the [OpenWithFragment] dialog.
+ * This will display browser apps and an item that can be used for installing Firefox,
+ * if it is not already installed on the device.
+ *
+ * @param infoArray List of browser apps.
+ * @param store Store app for installing Firefox.
+ */
+class AppAdapter(context: Context, infoArray: Array<ActivityInfo>, store: ActivityInfo?) :
+    RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    /**
+     * Class for an app installed on the device.
+     */
+    class App(private val context: Context, private val info: ActivityInfo) {
+        val label: String = info.loadLabel(context.packageManager).toString()
+
+        /**
+         * Retrieve the current graphical icon associated with the app.
+         */
+        fun loadIcon(): Drawable {
+            return info.loadIcon(context.packageManager)
+        }
+
+        val packageName: String
+            get() = info.packageName
+        val activityName: String
+            get() = info.name
+    }
+
+    /**
+     * Interface for setting a listener for an [App] item.
+     */
+    interface OnAppSelectedListener {
+        /**
+         * Action to be performed on an item from [AppAdapter] list being clicked.
+         */
+        fun onAppSelected(app: App)
+    }
+
+    private val apps: List<App>
```

Comment:
```suggestion
    private val apps: List<App> = infoArray.map { App(context, it) }
        .sortedWith { app1, app2 -> app1.label.compareTo(app2.label) }
```

Reply:
Thanks for the suggestion. I will keep initialisation in the init block, for visibility, but I will use the mapping suggestion, it works better! TY!